src/ui/App.tsx

```diff
@@ -103,29 +125,25 @@ function App(): JSX.Element {
 
   const viewStyle = {flex: 1};
 
-  const incrementThreadIndex = (): void => {
-    setThreadIndex(threadIndex + 1);
-  };
-
   const cards = [];
-  let numCardsRendered = 0;
-  while (
-    numCardsRendered < 10 &&
-    threadIndex + numCardsRendered < threads.length
-  ) {
-    const threadId = threads[threadIndex + numCardsRendered].id as string;
+  const threads = threadListState.threads;
+  // TODO: Once we take message fetching out of Card creation, only prerender
+  // one card below the most recently swiped card. Until then, render more cards
+  // and prevent rendering their messages to avoid getting stalled on slow
+  // network.
+  let numCardsRendered = 10;
+  while (numCardsRendered-- && numCardsRendered < threads.length) {
+    const threadId = threads[numCardsRendered].id as string;
     cards.push(
       <Card
         key={threadId}
         threadId={threadId}
         actions={threadActions}
-        cardSwipedAway={incrementThreadIndex}
+        cardSwipedAway={updateThreadListState}
```

Comment:
I'm confused here - what parameter gets passed to updateThreadListState?

Reply:
I now understand how this works (this is a prop, and the parameter is passed from within the Card) but this still feels a bit confusing.

I think I'd find it a bit clearer if we called this prop "updateThreadListState" or similar?

Reply:
Agree it's hard to follow, but from the Card's perspective it doesn't need to know anything about thread list state. Maybe rename it to onRemoveThread?